test/boost/sstable_compaction_test.cc

```diff
@@ -211,21 +211,25 @@ SEASTAR_TEST_CASE(compact) {
     table_for_tests cf(env.manager(), s);
     auto close_cf = deferred_stop(cf);
 
-        open_sstables(env, s, "test/resource/sstables/compaction", {1,2,3}).then([&env, s, cf, generation] (auto sstables) mutable {
+        auto sstables = open_sstables(env, s, "test/resource/sstables/compaction", {1,2,3}).get();
             auto new_sstable = [&env, gen = make_lw_shared<unsigned>(generation), s] {
                 return env.make_sstable(s, (*gen)++);
             };
-            return compact_sstables(sstables::compaction_descriptor(std::move(sstables), default_priority_class()), cf, new_sstable).then([&env, s, cf] (auto) {
+            compact_sstables(sstables::compaction_descriptor(std::move(sstables), default_priority_class()), cf, new_sstable).get();
                 // Verify that the compacted sstable has the right content. We expect to see:
                 //  name  | age | height
                 // -------+-----+--------
                 //  jerry |  40 |    170
                 //    tom |  20 |    180
                 //   john |  20 |   deleted
                 //   nadav - deleted partition
-                return env.reusable_sst(s, generation).then([&env, s] (shared_sstable sst) {
-                    auto reader = make_lw_shared<flat_mutation_reader_v2>(sstable_reader(sst, s, env.make_reader_permit())); // reader holds sst and s alive.
-                    return read_mutation_from_flat_mutation_reader(*reader).then([reader, s] (mutation_opt m) {
+                auto sst = env.reusable_sst(s, generation).get();
+                    auto reader = sstable_reader(sst, s, env.make_reader_permit());
+                    auto close_reader = deferred_close(reader);
+                    auto verify_mutation = [&] (std::function<void(mutation_opt)> verify) {
+                        std::invoke(verify, read_mutation_from_flat_mutation_reader(reader).get());
```

Comment:
Why std::invoke? You can write verify(read_mutation_from_flat_mutation_reader().get());. But don't bother changing.